From a95eebb24cbb56afd6fc40caa506ccea4e543b61 Mon Sep 17 00:00:00 2001
From: Scott Theisen <scott.the.elm@gmail.com>
Date: Fri, 19 Dec 2025 21:37:20 -0500
Subject: [PATCH] VAAPI deinterlacer: Use mode=default for DEINT_HIGH

FFmpeg says this will "Use the highest-numbered (and therefore
possibly most advanced) deinterlacing algorithm".

With my AMD Radeon RX 9060 XT, if deinterlacing is set to high
the VAAPI deinterlacer will not be created since it does not
support the motion_compensated deinterlacing mode.

By instead telling it to use default, it will select motion_adaptive.

Otherwise, see <va/va_vpp.h> and vaQueryVideoProcFilterCaps() to
actually get what VAAPI supports.
---
 mythtv/libs/libmythtv/opengl/mythvaapiinterop.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mythtv/libs/libmythtv/opengl/mythvaapiinterop.cpp b/mythtv/libs/libmythtv/opengl/mythvaapiinterop.cpp
index a2fbe83ae44..6a36cf5f1fc 100644
--- a/mythtv/libs/libmythtv/opengl/mythvaapiinterop.cpp
+++ b/mythtv/libs/libmythtv/opengl/mythvaapiinterop.cpp
@@ -227,7 +227,7 @@ bool MythVAAPIInterop::SetupDeinterlacer(MythDeintType Deinterlacer, bool Double
     if (DEINT_MEDIUM == Deinterlacer)
         deinterlacer = "motion_adaptive";
     else if (DEINT_HIGH == Deinterlacer)
-        deinterlacer = "motion_compensated";
+        deinterlacer = "default";
 
     // N.B. set auto to 0 otherwise we confuse playback if VAAPI does not deinterlace
     QString filters = QString("deinterlace_vaapi=mode=%1:rate=%2:auto=0")
